core/src/main/java/io/questdb/griffin/engine/functions/groupby/CountDistinctIntGroupByFunction.java

```diff
@@ -105,41 +101,95 @@ public String getName() {
         return "count_distinct";
     }
 
+    @Override
+    public int getValueIndex() {
+        return valueIndex;
+    }
+
     @Override
     public boolean isConstant() {
         return false;
     }
 
+    @Override
+    public boolean isParallelismSupported() {
+        return true;
+    }
+
     @Override
     public boolean isReadThreadSafe() {
         return false;
     }
 
+    @Override
+    public void merge(MapValue destValue, MapValue srcValue) {
+        long srcCount = srcValue.getLong(valueIndex);
+        if (srcCount == 0 || srcCount == Numbers.LONG_NaN) {
+            return;
+        }
+        long srcPtr = srcValue.getLong(valueIndex + 1);
+
+        long destCount = destValue.getLong(valueIndex);
+        if (destCount == 0 || destCount == Numbers.LONG_NaN) {
+            destValue.putLong(valueIndex, srcCount);
```

Comment:
This leaves an incorrect value at `valueIndex` slot, it is supposed to be the distinct count but it may be a non-distinct count of all the aggregated values if set like that.
I couldn't hit this path this the tests.

Reply:
Why do you think it sets an incorrect value? We know that the destination partial result haven't met any values while there are values in the source. In this case, we're free to replace the destination with the source. Am I missing anything?